plugins/tedge_apt_plugin/tests/main.rs

```diff
@@ -1,163 +1,172 @@
-use hamcrest2::prelude::*;
-use serial_test::serial;
-use std::process::{Command, Stdio};
-use std::sync::Once;
-use tedge_utils::fs::atomically_write_file_sync;
-use test_case::test_case;
-
-const TEDGE_APT_COMMAND: &str = "/etc/tedge/sm-plugins/apt";
-const APT_COMMAND: &str = "/usr/bin/apt-get";
-const PACKAGE_NAME: &str = "rolldice";
-const PACKAGE_VERSION: &str = "1.16-1+b3";
-
-#[cfg(target_arch = "x86_64")]
-const PACKAGE_URL: &str =
-    "http://ftp.br.debian.org/debian/pool/main/r/rolldice/rolldice_1.16-1+b3_amd64.deb";
-
-#[cfg(target_arch = "x86_64")]
-const PACKAGE_FILE_PATH: &str = "/tmp/rolldice_1.16-1+b3_amd64.deb";
-
-#[cfg(target_arch = "aarch64")]
-const PACKAGE_URL: &str =
-    "http://ftp.br.debian.org/debian/pool/main/r/rolldice/rolldice_1.16-1+b3_arm64.deb";
-
-#[cfg(target_arch = "aarch64")]
-const PACKAGE_FILE_PATH: &str = "/tmp/rolldice_1.16-1+b3_arm64.deb";
-
-#[cfg(target_arch = "arm")]
-const PACKAGE_URL: &str =
-    "http://ftp.br.debian.org/debian/pool/main/r/rolldice/rolldice_1.16-1+b3_armhf.deb";
-
-#[cfg(target_arch = "arm")]
-const PACKAGE_FILE_PATH: &str = "/tmp/rolldice_1.16-1+b3_armhf.deb";
-
-static DOWNLOAD_PACKAGE_BINARY: Once = Once::new();
-
-pub fn download_package_binary_once() {
-    DOWNLOAD_PACKAGE_BINARY.call_once_force(|_state| {
-        simple_download(PACKAGE_URL);
-    });
-}
-
-fn simple_download(url: &str) {
-    let response = reqwest::blocking::get(url).unwrap();
-    let content = response.bytes().unwrap();
-
-    atomically_write_file_sync("/tmp/rolldice.deb", PACKAGE_FILE_PATH, content.as_ref()).unwrap();
-}
-
-/// executes a `cmd` with `args`
-/// returns the stdout, stderr and exit code
-fn run_cmd(cmd: &str, args: &str) -> anyhow::Result<(String, String, i32)> {
-    let args = args.split_whitespace().collect::<Vec<&str>>();
-    let output = Command::new(cmd)
-        .args(args)
-        .stdin(Stdio::null())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .output()?;
-
-    let stdout = String::from_utf8(output.stdout).unwrap();
-    let stderr = String::from_utf8(output.stderr).unwrap();
-    let status_code = output.status.code().unwrap();
-    Ok((stdout, stderr, status_code))
-}
-
-// Parameters:
-//
-// input command
-//
-// expected stderr
-//
-// expected exit code
-//
-// description about the test
-#[test_case(
+#[cfg(feature = "integration-test")]
+mod tests {
+    use hamcrest2::prelude::*;
+    #[allow(unused_imports)]
```

Comment:
Instead of marking these as dead code, I believe you can mask these using the same `requires-sudo` feature gate that the test using this constant is gated with. Same for the `dead_code` annotations used below.